gh-137026: Add an explainer guide for asyncio (#137215)
python/cpython: Doc/library/asyncio.rst

asyncio is often a perfect fit for IO-bound and high-level
**structured** network code.

> If you're new to asyncio or confused by it and would like to better understand

Reviewer: This seems too informal to me, it can be as simple as "For a conceptual overview of asyncio checkout ..."

Author: This was updated in a slightly different vein per a different suggestion to use a `seealso` block which also features more formal language.

> If you're new to asyncio or confused by it and would like to better understand
> the fundmentals of how it works check out: :ref:`a-conceputal-overview-of-asyncio`.
> 

Reviewer 2: ```suggestion
.. seealso::

   HOWTO :ref:`a-conceputal-overview-of-asyncio`
      Explanation of the fundamentals of asyncio.
```

Reviewer 3: Little suggestion on your suggestion (which is a great suggestion)

```suggestion
.. seealso::

   HOWTO: :ref:`a-conceputal-overview-of-asyncio`
      Explanation of the fundamentals of asyncio.
```

Author: The "HOWTO" felt a bit out of place to me when looking at the rendered version. I removed the "HOWTO" but otherwise modified as suggested 👍

Author: I *think* this has been addressed, so I'm going to mark this as done.
